types/types.go

```diff
@@ -200,6 +200,23 @@ type ImageInspectInfo struct {
 	Layers        []string
 }
 
+// DockerAuthConfig contains authorization information for connecting to a registry.
+// FIXME: forked from docker/docker, waiting a full docker re-vendor
+type DockerAuthConfig struct {
+	Username string
+	Password string
+	Auth     string
+	// FIXME: currently not used
+	ServerAddress string
+	// IdentityToken is used to authenticate the user and get
+	// an access token for the registry.
+	// FIXME: currently not used
+	IdentityToken string
+	// RegistryToken is a bearer token to be sent to a registry
+	// FIXME: currently not used
+	RegistryToken string
```

Comment:
Ocid is going to be an user of this whole struct as soon as containers/image is plumbed there.. This is somehow copied from that..